pycontrails/core/vector.py

```diff
@@ -1010,6 +1064,35 @@ def create_empty(
         """
         return cls(data=_empty_vector_dict(keys or set()), attrs=attrs, copy=False, **attrs_kwargs)
 
+    @classmethod
+    def from_dict(
+        cls: Type[VectorDatasetType], obj: dict[str, Any], copy: bool = True
```

Comment:
You might also add an `**obj_kwargs` parameter to pass in additional k-v pairs as keyword args. The xarray API often allows both a dict parameter and a kwargs parameter -- it's kind of nice.

Reply:
f86373e2dfd5af37483e1bbb6dfa8b8af945e53f